### PR TITLE
Add support for setting bandwidth limit

### PR DESCRIPTION
**Pull request by kerneltime**
Merge after https://github.com/minio/minio/pull/10501 is merged and go.mod is updated.

### CODE REVIEW

File: cmd/admin-bucket-remote-add.go [hunk 1]
```diff
@@ -183,18 +189,22 @@ func fetchRemoteTarget(cli *cli.Context) (sourceBucket string, bktTarget *madmin
 	if !madmin.ServiceType(serviceType).IsValid() {
 		fatalIf(errInvalidArgument().Trace(serviceType), "Invalid service type. Valid option is `[replication]`.")
 	}
-
+	bandwidthStr := cli.String("bandwidth")
+	bandwidth, e := humanize.ParseBytes(bandwidthStr)
+	fatalIf(probe.NewError(e).Trace(bandwidthStr), "Unable to parse quota")
+	fmt.Printf("bandwidth:%d\n", bandwidth)
```
Comment (poornas) [inline cmd/admin-bucket-remote-add.go:195]: isn't this an optional flag? You might want to check if cli.IsSet("bandwidth") and set it in BucketTarget if this flag is set.

**Review by poornas — CHANGES_REQUESTED** [review]

File: cmd/admin-bucket-remote-add.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ import (
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/console"
 	"github.com/minio/minio/pkg/madmin"
+	"k8s.io/apimachinery/pkg/api/resource"
```
Comment (harshavardhana) [inline cmd/admin-bucket-remote-add.go:33]: I think we should use humanize package which we standardized for `mc` @kerneltime

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by poornas** [comment]
this new flag should be documented in the admin docs.

File: cmd/admin-bucket-remote-add.go (hunk 2 above)
Comment (kerneltime) [reply cmd/admin-bucket-remote-add.go]: Yup, moving back to humanize after discussion, as long as humanize is a superset of what kubernetes resources use for documentation it should be fine.

File: cmd/admin-bucket-remote-add.go [hunk 3]
```diff
@@ -78,12 +83,15 @@ FLAGS:
...
-                 https://foobar:foo12345@minio2:9000/replicabucket \
+  1. Set a new remote replication target 'targetbucket' in region "us-west-1" on https://minio.siteb.example.com for bucket 'sourcebucket'.
+     {{.Prompt}} {{.HelpName}} sitea/sourcebucket \
+                 https://foobar:foo12345@minio.siteb.example.com/targetbucket \
                  --service "replication" --region "us-west-1"
-     {{.EnableHistory}}
+  2. Set a new remote replication target 'targetbucket' in region "us-west-1" on https://minio.siteb.example.com for bucket 'sourcebucket' with bandwidth set to 2 gigabits (10^9) per second.
+     {{.Prompt}} {{.HelpName}} sitea/sourcebucket \
+                 https://foobar:foo12345@minio.siteb.example.com/targetbucket \
+                 --service "replication" --region "us-west-1 --bandwidth 2G"
```
Comment (harshavardhana) [inline cmd/admin-bucket-remote-add.go:93]:
```suggestion
                 --service "replication" --region "us-west-1 --bandwidth "2G"
```

File: cmd/admin-bucket-remote-add.go [hunk 4]
```diff
@@ -183,22 +192,40 @@ func fetchRemoteTarget(cli *cli.Context) (sourceBucket string, bktTarget *madmin
...
+		Region:         cli.String("region"),
+		BandwidthLimit: int64(bandwidth),
 	}
 	return sourceBucket, bktTarget
 }
 
+func getBandwidthInBytes(bandwidthStr string) (bandwidth uint64, err error) {
+	if bandwidthStr != "" {
+		bandwidthStr = strings.ToLower(bandwidthStr)
+		bandwidthStr = strings.TrimSuffix(bandwidthStr, "ps")
```
Comment (harshavardhana) [inline cmd/admin-bucket-remote-add.go:219]: why do we need to support this ? is it like Gbps ?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/admin-bucket-remote-add.go (hunk 4 above)
Comment (kerneltime) [reply cmd/admin-bucket-remote-add.go:219]: Yes

**Comment by kerneltime** [comment]
> this new flag should be documented in the admin docs.

Added

File: docs/minio-admin-complete-guide.md [hunk 5]
```diff
@@ -872,14 +872,28 @@ COMMANDS:
...
 
 ```
 mc admin bucket remote add myminio/srcbucket https://foobar:foobar12345@minio2:9000/targetbucket --service "replication" --region "us-west-1"
 ARN = `arn:minio:replication:us-west-1:1f8712ba-e38f-4429-bcb1-a7bb5aa97447:targetbucket`
 ```
 
+*Example: Add a new replication target `targetbucket` in region `us-west-1` on `https://minio2:9000` for bucket `srcbucket` on MinIO server. `foobar` and `foo12345` are credentials to target endpoint. The max bandwidth is metric 2G (2*10^9)*
+
+```
+mc admin bucket remote add myminio/srcbucket https://foobar:foobar12345@minio2:9000/targetbucket --service "replication" --region "us-west-1 --bandwidth 2G"
```
Comment (poornas) [inline docs/minio-admin-complete-guide.md:886]:
```suggestion
mc admin bucket remote add myminio/srcbucket https://foobar:foobar12345@minio2:9000/targetbucket --service "replication" --region "us-west-1" --bandwidth "2G"
```

File: docs/minio-admin-complete-guide.md [hunk 6]
```diff
@@ -872,14 +872,28 @@ COMMANDS:
...
+ARN = `arn:minio:replication:us-west-1:1f8712ba-e38f-4429-bcb1-a7bb5aa97447:targetbucket`
+```
+
+*Example: Add a new replication target `targetbucket` in region `us-west-1` on `https://minio2:9000` for bucket `srcbucket` on MinIO server. `foobar` and `foo12345` are credentials to target endpoint. The max bandwidth is IEC 2Gi (2*2^30)*
+
+```
+mc admin bucket remote add myminio/srcbucket https://foobar:foobar12345@minio2:9000/targetbucket --service "replication" --region "us-west-1 --bandwidth 2Gi"
```
Comment (poornas) [inline docs/minio-admin-complete-guide.md:893]:
```suggestion
mc admin bucket remote add myminio/srcbucket https://foobar:foobar12345@minio2:9000/targetbucket --service "replication" --region "us-west-1" --bandwidth "2Gi"
```

**Review by poornas — CHANGES_REQUESTED** [review]

**Review by poornas — APPROVED** [review]
LGTM

**Review by BigUstad — APPROVED** [review]
LGTM

**Pull request merged**